sys/include/net/sock/addr.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *                    Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_sock_addr   Address abstractions
+ * @ingroup     net_sock
+ * @brief       Address abstractions for usage with @ref net_sock
+ * @{
+ *
+ * @file
+ * @brief   Address abstraction definitions for @ref net_sock
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ * @author  Kaspar Schleiser <kaspar@schleiser.de>
+ */
+#ifndef SOCK_ADDR_H_
+#define SOCK_ADDR_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Special @ref net_netif "netif" ID for "any interface"
+ * @todo    Define in @ref net_netif
+ */
+#define SOCK_ADDR_ANY_NETIF (0)
+
+/**
+ * @brief   Address to bind to any IPv4 address
+ */
+#define SOCK_ADDR_IPV4_ADDR_ANY (0U)
+
+/**
+ * @brief   Address to bind to any IPv6 address
+ */
+#define SOCK_ADDR_IPV6_ADDR_ANY { { 0 } }
+
+typedef uint32_t sock_addr_ipv4_t;
+
+/**
+ * @brief   Type to represent an IPv6 address
+ */
+typedef struct {
+    uint8_t u8[16];         /**< bytes of the address */
+} sock_addr_ipv6_t;
```

Comment:
> In 6lowpan, an address might be at an unaligned position in the header.

What does this have to do with the data type for the IPv6 address?

> That's one network stack (gnrc) for one platform (RIOT). Both emb6 and lwip provide their own structs. POSIX has another way of storing IP addresses (sockaddr_*). When discussing this, we decided to abstract this at the user API level.

The data struct for IPv6 address is GNRC independent (for a good reason). That  packages define their own redundant data types is inevitable and bound to the nature of a package. Defining redundant data types inside RIOT is IMO confusing.

> When discussing this, we decided to abstract this at the user API level.

Why is `sys/include/net/sock/addr.h` more user API level than `sys/include/net/ipv6/addr.h`?

Sorry, Kaspar, none of your response make any sense to me. Maybe I miss a point.